src/ax/saga/persistence/eventsourcing/persister.go

```diff
@@ -66,23 +70,24 @@ func (p *Persister) BeginUnitOfWork(
 		return nil, err
 	}
 
-	return &unitOfWork{
-		p.MessageStore,
-		p.Snapshots,
-		p.SnapshotFrequency,
-		tx,
-		pk,
-		&Recorder{Next: s},
-		i,
-	}, nil
+	uow.messageStore = p.MessageStore
+	uow.snapshots = p.Snapshots
+	uow.frequency = p.SnapshotFrequency
+	uow.tx = tx
+	uow.key = pk
+	uow.recorder = &Recorder{Next: s}
+	uow.instance = i
+
+	return uow, nil
 }
 
 // unitOfWork is an implementation of saga.UnitOfWork that perists saga
 // instances as a stream of events, with optional snapshots.
 type unitOfWork struct {
-	messageStore messagestore.Store
-	snapshots    SnapshotRepository
-	frequency    saga.Revision
+	messageStore      messagestore.Store
+	snapshots         SnapshotRepository
+	frequency         saga.Revision
+	lastKnownSnapshot saga.Revision
```

Comment:
<super picky> Can you move `lastKnownSnapshot` to the second "block" of fields. The first block are the dependencies inherited from the `Persister`, the second block are request-specific fields. There is no way you could have known this :P